File: java/src/main/java/it/giannibombelli/wsc2026/booking/domain/ports/BookingRepository.java
package it.giannibombelli.wsc2026.booking.domain.ports;

import it.giannibombelli.wsc2026.booking.domain.booking.Booking;
import it.giannibombelli.wsc2026.booking.domain.booking.BookingId;

import java.util.Optional;

public interface BookingRepository {
    void save(Booking booking);

    Optional<Booking> findById(BookingId id);
}

File: typescript/src/booking/domain/ports/bookingRepository.ts
// Porta per la persistenza degli aggregati Booking.

import type { Booking } from '../booking/booking.js';
import type { BookingId } from '../booking/bookingId.js';

export interface BookingRepository {
  save(booking: Booking): void;

  findById(id: BookingId): Booking | null;
}
